src/ucp/core/ucp_listener.c

```diff
@@ -22,128 +22,38 @@
 
 static unsigned ucp_listener_accept_cb_progress(void *arg)
 {
-    ucp_ep_h       ep       = arg;
-    ucp_listener_h listener = ucp_ep_ext_control(ep)->listener;
+    ucp_conn_request_h conn_request = arg;
+    ucp_listener_h listener         = conn_request->listener;
+    ucp_ep_h ep                     = conn_request->ep;
 
-    /* NOTE: protect union */
-    ucs_assert(!(ep->flags & (UCP_EP_FLAG_ON_MATCH_CTX |
-                              UCP_EP_FLAG_FLUSH_STATE_VALID)));
-    ucs_assert(ep->flags   & UCP_EP_FLAG_LISTENER);
+    ucs_free(conn_request->remote_dev_addr);
+    ucs_free(conn_request);
```

Comment:
looks weird, if we don't use it why needed to allocate it?

Reply:
we use it in `ucp_ep_create_server_accept` which is "preferred" path to create the ep